tests/operator/helper.go

```diff
@@ -45,3 +53,97 @@ func SplitCsv(csv string) CsvResult {
 	}
 	return result
 }
+
+func getAllPodsBy(namespace string) (podsInNamespace []*provider.Pod) {
+	for i := range env.AllPods {
+		pod := env.AllPods[i]
+		if pod.Namespace == namespace {
+			podsInNamespace = append(podsInNamespace, pod)
+		}
+	}
+	return podsInNamespace
+}
+
+func checkValidOperatorPodsIn(namespace string, allPods []*provider.Pod) (foundCsvs map[string]bool, podsBelongingToNoOperators []string, err error) {
+	foundCsvs = make(map[string]bool)
+	for index := range allPods {
+		pod := allPods[index]
+		topOwners, err := podhelper.GetPodTopOwner(pod.Namespace, pod.OwnerReferences)
+		if err != nil {
+			return foundCsvs, podsBelongingToNoOperators, err
+		}
+
+		validOwnerFound := false
+		for _, owner := range topOwners {
+			if owner.Kind == v1alpha1.ClusterServiceVersionKind && owner.Namespace == namespace {
+				foundCsvs[owner.Name] = true
```

Comment:
We may need more checks here. other valid top owner could be the operands created based on the crds installed by this operator. Actually these operands can only be installed in the single operator namespace. For instance, for prometheus, pods belonging to the prometheus object top owner should be valid. We could list all the crds installs by this operator by reading the crd label: "operatorframework.io/installed-alongside-..." or similar then use the crd name to identify valid top owners.

Reply:
@edcdavid that's only valid for operators installed with the [ownNamespace](https://olm.operatorframework.io/docs/advanced-tasks/operator-scoping-with-operatorgroups/#ownnamespace) mode. For [singleNamespace](https://olm.operatorframework.io/docs/advanced-tasks/operator-scoping-with-operatorgroups/#singlenamespace) operators, the controller is running in namespace **A** and watches for CRs in namespace **B**, so, in theory, operand pods should only be found in namespace **B**.

The problem is the requirement only referes to SingleNamespaced and AllNamespaced operators... so what about ownNamespaced and multiNamespaced ones?

Reply:
ah, you are right about single namespace vs own namespace, in single namespace only mode, the operand should be in only in namespace B, sorry for the confusion. So in this case, we still need a way to ignore the csv that will be created in namespace B otherwise it will be picked up as an operator namespace.

Reply:
> The problem is the requirement only referes to SingleNamespaced and AllNamespaced operators... so what about ownNamespaced and multiNamespaced ones?
In my opinion, the multi namespace is similar to the single namespace and it would make sense to support it here.